train-tracker/src/functionality/app.js

```diff
@@ -1,14 +1,23 @@
-export function filterTrains(allTrains, selectedNumber, selectedRoute, selectedStation, upcoming, fromStation, toStation) {
-    let trains = allTrains;
-    if (selectedNumber > 0){
-        trains = trains.filter(t => t.number === selectedNumber)
-    } if (selectedRoute){
-        trains = trains.filter(t => t.routeName === selectedRoute)
-    } if (selectedStation){
-        trains = trains.filter((t) => (t.stations.findIndex((station) => station.stationCode === selectedStation) !== -1));
-        if (upcoming){
+// returns a copy of SearchObject with a property changed
+// react doesn't recognize just changing an object property
+// so we need to set it to a copy w/ the prop changed
+export function setProp(searchObj,prop,value){
```

Comment:
You could simplify this function to a one liner using the spread operator:

```
export function setProp(searchObj, prop, value) {
    return { ...searchObj, [prop]: value };
}
```

Then, we could probably get rid of this function entirely and just do this wherever that function is called originally? Obviously would have the same function, but would increase readability and be more in line with the way updating object state usually looks in React.

Reply:
cool one liner, I had no clue the spread operator existed